src/tests/test_double_compare.cpp

```diff
@@ -0,0 +1,220 @@
+#include <boost/math/special_functions/next.hpp>
+#include <gtest/gtest.h>
+
+#include "types.h"
+
+namespace tests {
+
+class DoubleCompare : public ::testing::Test {
+private:
+    std::unique_ptr<std::byte[]> double_min_ptr;
+    std::unique_ptr<std::byte[]> double_max_ptr;
+    std::unique_ptr<std::byte[]> result_ptr;
+
+protected:
+    model::DoubleType double_type;
+    model::INumericType& double_type_ref = double_type;
+
+    std::byte* double_min;
+    std::byte* double_max;
+    std::byte* result;
+
+    void SetUp() override {
+        double_min_ptr.reset(double_type.MakeValue(std::numeric_limits<model::Double>().min()));
+        double_max_ptr.reset(double_type.MakeValue(std::numeric_limits<model::Double>().max()));
+        result_ptr.reset(double_type.MakeValue(0.0));
+        result = result_ptr.get();
+        double_min = double_min_ptr.get();
+        double_max = double_max_ptr.get();
+    }
+};
+
+TEST_F(DoubleCompare, AddMin) {
+    std::unique_ptr<std::byte[]> number_128_ptr(double_type.MakeValue(128.0));  // 2^7
+    std::unique_ptr<std::byte[]> number_100_ptr(double_type.MakeValue(100.0));
+    std::unique_ptr<std::byte[]> buff_ptr(double_type.MakeValue(0));
+    std::unique_ptr<std::byte[]> less_ptr(double_type.MakeValue(0));
+
+    std::byte* buff = buff_ptr.get();
+    std::byte* number_128 = number_128_ptr.get();
+    std::byte* number_100 = number_100_ptr.get();
+    std::byte* less = less_ptr.get();
+
+    double_type_ref.Mul(number_128, double_min, result);
+    double_type_ref.Mul(number_100, double_min, less);
+
+    for (int i = 0; i < 128; i++) {
+        double_type_ref.Add(buff, double_min, buff);
+    }
+
+    ASSERT_EQ(double_type_ref.Compare(buff, result), model::CompareResult::kEqual);
+    ASSERT_EQ(double_type_ref.Compare(less, buff), model::CompareResult::kLess);
+}
+
+TEST_F(DoubleCompare, DivMax) {
+    std::unique_ptr<std::byte[]> three_ptr(double_type.MakeValue(3.0));
+    std::unique_ptr<std::byte[]> three_power_20_ptr(double_type.MakeValue(3486784401.0));
```

Comment:
```suggestion
    std::unique_ptr<std::byte[]> three_power_20_ptr(double_type.MakeValue(std::pow(3., 20.)));
```
And twenty can be stored in a variable `power` since it's also used later in the loop condition